### PR TITLE
Added documentation for profile tooling

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

This change added a few sections describing the process of profile CSV creation and how to integrate them into the platform build process.

In addition, this change also adds the documentation for newly introduced profile name and ID. Specifically, the data flow in firmware and platform integration steps are described in the doc.

Resolves https://github.com/microsoft/mu_feature_config/issues/241

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [x] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

N/A. Documentation change only.

## Integration Instructions

N/A

### CODE REVIEW

File: SetupDataPkg/Docs/Profiles/Overview.md [hunk 1]
```diff
@@ -99,3 +99,48 @@ knobs that can be iterated through and applied to the `CacheValueAddress` of eac
...
+
+If platforms would like to integrate the profile CSV files into their build, they can add the following to their build by
+following the statements outlined in the [section above](#flow).
+
+#### Provide Profile Names and IDs
+
+Platforms also can provide a 2-character name and/or 1-byte ID for each profile. These are used to identify the profile
+by the selector, which could be BMC, OS, or other.
+
+In firmware, the name and ID are provided in the `gProfileFlavorNames` and `gProfileFlavorIds`, respectively. Platforms
```
Comment (apop5) [inline SetupDataPkg/Docs/Profiles/Overview.md:135]:
```suggestion
In firmware, the name and ID are auto generated and can be accessed through `gProfileFlavorNames` and `gProfileFlavorIds`, respectively. Platforms

```

File: SetupDataPkg/Docs/Profiles/Overview.md [hunk 2]
```diff
@@ -99,3 +99,48 @@ knobs that can be iterated through and applied to the `CacheValueAddress` of eac
...
+can elect to create HOBs for these meta data information and provide them to the selector. See example [here](https://github.com/microsoft/mu_oem_sample/tree/release/202302/OemPkg/OemConfigPolicyCreatorPei).
```
Comment (apop5) [inline SetupDataPkg/Docs/Profiles/Overview.md:136]:
```suggestion
can elect to create HOBs for this meta data information and consume it in other drivers. See example [here](https://github.com/microsoft/mu_oem_sample/tree/release/202302/OemPkg/OemConfigPolicyCreatorPei).

```

**Review by apop5 — APPROVED** [review]

File: SetupDataPkg/Docs/Profiles/Overview.md [hunk 3]
```diff
@@ -99,3 +99,48 @@ knobs that can be iterated through and applied to the `CacheValueAddress` of eac
...
+
+    ![Profile Through ConfigEditor UI tool](Images/profile_ui.png)
+
+- Through the command line:
+
+  - Run the following command:
+
+    ```bash
+    GenNCCfgData  GENCSV  XmlFile[;BinFile]   CsvOutFile
+    ```
```
Comment (makubacki) [inline SetupDataPkg/Docs/Profiles/Overview.md:123]: I think the command is simple enough, but you might also want to consider adding an actual example while you're here.

**Review by makubacki — APPROVED** [review]

File: SetupDataPkg/Docs/Profiles/Overview.md (hunk 3 above)
Comment (kuqin12) [reply SetupDataPkg/Docs/Profiles/Overview.md]: Thanks, updated.

**Pull request merged**